packages/compass-components/src/components/file-input.tsx

```diff
@@ -0,0 +1,180 @@
+/** @jsx jsx */
+import React from 'react';
+import path from 'path';
+import { css, jsx } from '@emotion/react';
+
+import { Button, Icon } from '..';
+
+const formItemHorizontalStyles = css`
+  margin: 15px auto 15px;
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const formItemVerticalStyles = css`
+  margin: 5px auto 20px;
+`;
+
+const formItemErrorStyles = css`
+  border: 1px solid red;
+
+  &:focus {
+    border: 1px solid red;
+  }
+`;
+
+const buttonStyles = css`
+  width: 100%;
+`;
+
+const buttonErrorStyles = css`
+  border: 1px solid red;
+`;
+
+const labelHorizontalStyles = css`
+  width: 70%;
+  text-align: right;
+  padding-right: 15px;
+  margin: auto;
+`;
+
+const labelErrorStyles = css`
+  color: red;
+`;
+
+const labelIconStyles = css`
+  display: inline-block;
+  font: normal normal normal 14px/1 FontAwesome;
+  font-size: inherit;
+  text-rendering: auto;
+  margin: 0 0 0 5px;
+  cursor: pointer;
+  color: #bfbfbe;
+
+  &:link,
+  &:active {
+    color: #bfbfbe;
+  }
+
+  &:link,
+  &:active,
+  &:hover {
+    text-decoration: none;
+  }
+
+  &:hover {
+    color: #fbb129;
+  }
+`;
+
+export enum Variant {
+  Horizontal = 'HORIZONTAL',
+  Vertical = 'VERTICAL',
+}
+
+// https://www.electronjs.org/docs/latest/api/file-object
+interface FileWithPath extends File {
```

Comment:
This makes sense. You can also to just `type FileWithPath = File & { path: string }`. There was a point in the past where making interfaces was considered a better practice in TS because `type` was pretty limited, but I don't think it's that important anymore. Another thing just FYI, you can also have a comment that describes the prop near it so it shows up in the intellisense